core/templates/components/skill-selector/skill-selector.component.ts

```diff
@@ -0,0 +1,213 @@
+        // If no filter is applied on both subtopics and topics, we
+        // need to display all the skills (the original list).
+        this.currCategorizedSkills = angular.copy(this.categorizedSkills);
+      }
+    } else {
+      this.currCategorizedSkills = angular.copy(updatedSkillsDict);
+    }
+  }
+
+  // The folowing function is called when the topic filter changes.
+  // First, the subtopic filter is updated according to the changed
+  // topic filter list. Then the main Skills list is updated.
+  updateSkillsListOnTopicFilterChange(): void {
+    var updatedSubTopicFilterList = {};
+    var isAnyTopicChecked = false;
+    for (var i = 0; i < this.topicFilterList.length; i++) {
+      if (this.topicFilterList[i].checked) {
+        var topicName = this.topicFilterList[i].topicName;
+        updatedSubTopicFilterList[topicName] = (
+          angular.copy(this.intialSubTopicFilterDict[topicName]));
+        isAnyTopicChecked = true;
+      }
+    }
+    if (!isAnyTopicChecked) {
+      // If there are no topics checked on topic filter, we have to
+      // display subtopics from all the topics in the subtopic filter.
+      for (var topic in this.intialSubTopicFilterDict) {
+        if (!this.subTopicFilterDict.hasOwnProperty(topic)) {
+          this.subTopicFilterDict[topic] = (
+            angular.copy(this.intialSubTopicFilterDict[topic]));
+        }
+      }
+    } else {
+      this.subTopicFilterDict =
+         angular.copy(updatedSubTopicFilterList);
+    }
+    // After we update the subtopic filter list, we need to update
+    // the main skills list.
+    this.updateSkillsListOnSubtopicFilterChange();
+  }
+
+  searchInSubtopicSkills(input: ShortSkillSummary[], searchText: string):
+  ShortSkillSummary[] {
+    let skills: string[] = input.map(val => {
+      return val.getDescription();
+    });
+    let filteredSkills = this.filterForMatchingSubtringPipe
+      .transform(skills, searchText);
+    return input.filter(val => {
+      return filteredSkills.includes(val.description);
+    });
+  }
+
+  searchInUntriagedSkillSummaries(searchText: string): SkillSummary[] {
+    let skills: string[] = this.untriagedSkillSummaries.map(val => {
+      return val.description;
+    });
+    let filteredSkills = this.filterForMatchingSubtringPipe
+      .transform(skills, searchText);
+    return this.untriagedSkillSummaries.filter(val => {
+      return filteredSkills.includes(val.description);
+    });
+  }
+
+  clearAllFilters(): void {
+    for (var i = 0; i < this.topicFilterList.length; i++) {
+      this.topicFilterList[i].checked = false;
+    }
+    for (let topicName in this.subTopicFilterDict) {
+      var length = this.subTopicFilterDict[topicName].length;
+      for (var j = 0; j < length; j++) {
+        this.subTopicFilterDict[topicName][j].checked = false;
+      }
+    }
+    this.updateSkillsListOnTopicFilterChange();
+  }
+}
+
+angular.module('oppia').directive(
+  'skillSelector', downgradeComponent(
```

Comment:
change to `oppiaSkillSelector`

Reply:
I think this should be fine since the component is `skill-selector` -- will see what decision is made in above comment.